sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -1214,3 +1214,22 @@ def test_linear_models_cv_fit_for_all_backends(backend, estimator):
 
     with joblib.parallel_backend(backend=backend):
         estimator(n_jobs=2, cv=3).fit(X, y)
+
+
+@pytest.mark.parametrize("check_input", [True, False])
+def test_enet_sample_weight_does_not_overwrite_sample_weight(check_input):
+    """Check that elastic next does not overwrite sample_weights."""
+
+    rng = np.random.RandomState(0)
+    n_samples, n_features = 10, 5
+
+    X = rng.rand(n_samples, n_features)
+    y = rng.rand(n_samples)
+
+    sample_weight_1_25 = 1.25 * np.ones_like(y)
+    sample_weight = sample_weight_1_25.copy()
+
+    reg = ElasticNet()
+    reg.fit(X, y, sample_weight=sample_weight, check_input=check_input)
+
+    assert_allclose(sample_weight, sample_weight_1_25)
```

Comment:
Could we even test with `assert_array_equal(sample_weight, sample_weight_1_25)`?

Reply:
We tend to avoid checking strict equality between floats to avoid failures due to rounding errors, although here it should hold since sample_weight should not be modified at all. I guess it doesn't hurt to be extra cautious.